lxd/main_cluster.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	lxd "github.com/lxc/lxd/client"
+	"github.com/lxc/lxd/lxd/cluster"
+	"github.com/lxc/lxd/lxd/db"
+	"github.com/lxc/lxd/lxd/sys"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/i18n"
+	"github.com/olekukonko/tablewriter"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+type cmdCluster struct {
+	global *cmdGlobal
+}
+
+func (c *cmdCluster) Command() *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = "cluster"
+	cmd.Short = "Low-level cluster administration commands"
+	cmd.Long = `Description:
+  Low level administration tools for inspecting and recovering LXD clusters.
+`
+	// List database nodes
+	listDatabaseNodes := cmdClusterListDatabaseNodes{global: c.global}
+	cmd.AddCommand(listDatabaseNodes.Command())
+
+	// Recover
+	recover := cmdClusterRecoverFromQuorumLoss{global: c.global}
+	cmd.AddCommand(recover.Command())
+
+	return cmd
+}
+
+type cmdClusterListDatabaseNodes struct {
+	global *cmdGlobal
+}
+
+func (c *cmdClusterListDatabaseNodes) Command() *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = i18n.G("list-database-nodes")
+	cmd.Aliases = []string{"ls"}
+	cmd.Short = i18n.G("Print the addresses of the cluster members serving as database nodes")
+
+	cmd.RunE = c.Run
+
+	return cmd
+}
+
+func (c *cmdClusterListDatabaseNodes) Run(cmd *cobra.Command, args []string) error {
+	os := sys.DefaultOS()
+
+	db, _, err := db.OpenNode(filepath.Join(os.VarDir, "database"), nil, nil)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to open local database.")
+	}
+
+	addresses, err := cluster.ListDatabaseNodes(db)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get database nodes.")
+	}
+
+	printDatabaseNodes(addresses)
```

Comment:
Doesn't feel like there's a good reason to split this into its own function given it's only used here.

One idea would be to have a commit which moves `renderTable` from `lxc/utils.go` to a `lxc/utils/list.go` and then use that in the `lxd` command too, effectively having the exact same rendering logic and also allowing dumping as csv/yaml/json/...

Reply:
The only reason is to avoid the clash of the variable name ```os``` with the package ```os```, but I can change the variable name if you prefer.

Reply:
Feels like moving `renderTable` over to `lxc/utils/`, exporting it and using it in `lxc` and `lxd` is probably the best thing to do here.

Reply:
Done